### PR TITLE
test: sed on FreeBSD requires "-i extension", so use gsed

### PR DESCRIPTION
**Pull request by wjwithagen**
- FreeBSD sed wants a backup extention after the -i option
   or otherwise it starts eating the next option

Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

File: src/test/crush/crush-classes.sh [hunk 1]
```diff
@@ -71,7 +71,12 @@ function TEST_classes() {
     #
     ceph osd getcrushmap > $dir/map || return 1
     crushtool -d $dir/map -o $dir/map.txt || return 1
-    sed -i \
+    if [ `uname` = FreeBSD ]; then
```
Comment (tchaikov) [inline src/test/crush/crush-classes.sh]: how about `src/test/mon/osd-crush.sh` , `src/test/test_pool_create.sh` and `src/test/test_crush_bucket.sh`, maybe we can define a function in `qa/workunits/ceph-helpers.sh` to normalize the difference between FreeBSD sed and GNU sed?

**Comment by wjwithagen** [comment]
@tchaikov 
It is certainly wobbly argument passing on behalf of FreeBSD.
And using more than one `-e` set is a way to trigger.
But I'll look into "generalizing" this.

**Comment by tchaikov** [comment]
FreeBSD sed
```
sed [-Ealn] [-e command] [-f command_file] [-i extension] [file ...]
```
GNU sed
```
-i[SUFFIX], --in-place[=SUFFIX]
```

@wjwithagen i don't think it's triggered by `-e`, but yeah. we need to generalize this. because otherwise we need to patch multiple test script in the very same way.

**Comment by wjwithagen** [comment]
@tchaikov 
Sorry, but I've been rather busy with other things lately.
But once the dust settles here, I try and fix this thru `qa/workunits/ceph-helpers.sh`

**Comment by wjwithagen** [comment]
@tchaikov 
I'm looking into `ceph-helpers.sh` but it would be the first "repair" function this way.
Only thing fixing there is the `xmlstarlet` <>`xml` compatibility?

Are you sure you'd like to do it this way?
ATM `crush-classes.sh` is the only where it does not work as expected.
So for the time being I'd keep this patch, and I'll look into the FreeBSD code of sed and/or see if we can start using the GNU version of sed.

**Comment by tchaikov** [comment]
@wjwithagen 

take 
https://github.com/ceph/ceph/blob/master/src/test/mon/osd-crush.sh#L149 for example, are you sure FreeBSD works as expected? i believe it will take `'s/ruleset 0/ruleset 3/'` as the option of the `-i` argument.

**Comment by wjwithagen** [comment]
@tchaikov 
I chose the easier way out: use GNU sed

**Comment by tchaikov** [comment]
> Are you sure you'd like to do it this way?

@wjwithagen i don't really like this way. and yeah! i like the "gsed" approach more.

**Comment by wjwithagen** [comment]
@tchaikov 
Jenkins seems happy also.
So you might want to run it thru QA?

**Comment by tchaikov** [comment]
@wjwithagen will do.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-03-20_17:19:01-rados-wip-kefu-testing---basic-smithi/
- http://pulpito.ceph.com/kchai-2017-03-23_06:49:46-rados-wip-kefu-testing---basic-mira/

**Pull request merged**